castor-service/src/main/java/io/carbynestack/castor/service/persistence/cache/ReservationCachingService.java

```diff
@@ -146,13 +194,70 @@ public void keepAndApplyReservation(Reservation reservation) {
           fragment.setReservationId(reservation.getReservationId());
           tupleChunkFragmentStorageService.update(fragment);
           startIndex = fragment.getEndIndex();
+
+        } catch (Exception e) {
+          System.err.println(
+              "Reservation of single fragment failed; probably locking issue; res:"
+                  + reservation.getReservationId()
+                  + " tchunk:"
+                  + re.getTupleChunkId());
+          throw e;
+        }
+      }
+    }
+    if (firstRoundFragment < reservation.getReservations().size()) {
+
+      HashMap<UUID, ArrayList<Long>> mappedReservations = new HashMap<>();
+      List<ReservationElement> roundReservationElements =
+          reservation
+              .getReservations()
+              .subList(firstRoundFragment, reservation.getReservations().size());
+      roundReservationElements.forEach(
+          resElem -> {
+            mappedReservations.computeIfAbsent(resElem.getTupleChunkId(), k -> new ArrayList<>());
+            mappedReservations.get(resElem.getTupleChunkId()).add(resElem.getStartIndex());
+          });
+
+      for (UUID tChunkId : mappedReservations.keySet()) {
+        int retryMillisecs = 1500;
+        while (true) {
+          try {
+            int actuallyReservedFragments =
+                tupleChunkFragmentStorageService.reserveRoundFragmentsByIndices(
+                    mappedReservations.get(tChunkId), reservation.getReservationId(), tChunkId);
+            if (actuallyReservedFragments == 0) {
+              retryMillisecs -= 30;
```

Comment:
why not configurable, why no exponential backoff or other strategies? does the potential load make sense?